### ISSUE TITLE
Add custom commits to dependabot PRs

### ISSUE THREAD

**Issue by greysteil** [dependabot/dependabot-core#402]
_From @deivid-rodriguez on January 25, 2018 16:6_

Not sure if this is feasible, or whether it belongs in dependabot's scope, but I've missed something like this a few times so I thought I might propose it.

Many times I want to manually edit dependabot PR's. For example:

* If dependabot managed our `yarn.lock` file in [decidim](https://github.com/decidim/decidim), we would need to edit every PR to run `yarn run build:prod` and commit the result, since the way we use `yarn` there is to generate and vendor a `bundle.js` asset file using `webpack`.

* When I update [image-optim-pack](https://github.com/toy/image_optim_pack), I always want to run `bundle exec image-optim app/assets/images` and commit new optimizations in the new version of the gem.

* When I bump rubocop, I usually want to run `bundle exec rubocop --auto-correct` and commit the new style fixes.

I wonder if a list of commands could be configurable and added as extra commits for each PR if running them lead to changes in source controlled files.

_Copied from original issue: dependabot/feedback#78_

**Comment by greysteil** [comment]
Sorry for the slow reply on this one - I've been mulling it over.

I think doing the above as a pipeline of commands to be run is definitely the way forward, but the way Dependabot is currently set up means it can't support them. In particular, because we pull down only the dependency files, rather than cloning repos, running any scripts and committing the results is likely to cause problems. It's not insurmountable, but it makes things more difficult.

Outside of Dependabot's core concerns, I wonder if there's a way to do this with a CI, in the same way that [greenkeeper lockfile](https://github.com/greenkeeperio/greenkeeper-lockfile) works. I've always thought that was a bad solution for lockfiles, but it's not a bad solution for custom flows. Perhaps we could put together a similar, more generic set of code snippets for people to paste into their CI setup.

So, TL;DR - I don't see support for running generic scripts being added to core Dependabot anytime soon, but I'd be up for devising and promoting a workaround.

**Comment by greysteil** [comment]
_From @deivid-rodriguez on January 31, 2018 15:41_

Thanks!
